### PR TITLE
csm: modify o11y test to use proxyless bootstrap injector

### PR DESCRIPTION
**Pull request by arvindbr8**
As part of CSM Gateway for Mesh GA - we have already implemented a Proxyless Bootstrap Injector in GCP which will allow clusters that are onboarded to CSM Gateway for Mesh to get the following injected into the pod with the right labels:

1. initContainer with TD Bootstrap Injector + volume Mount where the config output is generated
2. CSM O11y related labels and envvars. 

The test cluster also had to be updated to follow the new onboarding steps. I will create a CL internally to update the cluster creation script

Testing: [here](http://sponge2/659c620f-02a2-4cca-aafe-0864d2e9e2b2)

### CODE REVIEW

**Comment by arvindbr8** [comment]
Testing the new cluster + test driver [here](http://sponge2/659c620f-02a2-4cca-aafe-0864d2e9e2b2)

**Review by arvindbr8 — COMMENTED** [review]
The last one failed for connectivity issues. I was able to verify that the injector added the CSM o11y env vars and also the bootstrap config to the volume mount. 

Kicking off another run of the test suite:

- [ ] [grpc/core/master/linux/psm-csm-python](http://fusion2/e93ea3e0-1ca1-411f-99e8-3bbee32e91f6)

**Review by sergiitk — CHANGES_REQUESTED** [review]
Ideally I'd like to avoid creating new templates, and just configure everything with deployment args. Let's talk IRL about how we can achieve that.

**Comment by arvindbr8** [comment]
[latest run](https://fusion2.corp.google.com/invocations/849a8ecc-f061-49d9-ba56-45e9e5db633f/targets/grpc%2Fcore%2Fmaster%2Flinux%2Fpsm-csm;config=default/log) after modifying api version

File: tests/gamma/csm_observability_test_with_injection.py [hunk 1]
```diff
@@ -0,0 +1,627 @@
+# Copyright 2023 gRPC authors.
```
Comment (sergiitk) [inline tests/gamma/csm_observability_test_with_injection.py:1]:
```suggestion
# Copyright 2024 gRPC authors.
```

File: kubernetes-manifests/gamma/server.deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      serviceAccountName: ${service_account_name}
+      % endif
+      % if termination_grace_period_seconds:
+      terminationGracePeriodSeconds: ${termination_grace_period_seconds}
+      % endif
+      containers:
+        - name: ${deployment_name}
+          image: ${image_name}
+          imagePullPolicy: Always
+          # TODO:(lsafran) test without this when ipv6 HC fw rules are in place
```
Comment (sergiitk) [inline kubernetes-manifests/gamma/server.deployment.yaml:34]: update this so it doesn't render

**Review by sergiitk — APPROVED** [review]
LGTM if full CSM suite passes

**Comment by arvindbr8** [comment]
Running the entire suite again - 
- [ ] [grpc/core/v1.68.x/branch/linux/psm-csm](https://source.cloud.google.com/results/invocations/3e3a284a-857d-4a1a-9591-d78efff9008e)

**Pull request merged**